### PR TITLE
Fix a small bug at `rng` arg of mcmc

### PR DESCRIPTION
**Pull request by fehiepsi**
If users provide `rng` to `mcmc` with single chain, the current code will throw error because it expected that `rng` will have `batch_size = 1`. This bug comes from my last rng PR and this PR fixes that.

In addition, I
+ move `log_density` to `infer_util` because it is not specific to `hmc`,
+ move euclidean ke to `hmc_util`
+ fix [tutorial](http://pyro.ai/numpyro/bayesian_regression.html#Bayesian-Regression-Using-NumPyro) not render list of items. We need an empty line before each markdown list.

### CODE REVIEW

File: numpyro/mcmc.py [hunk 1]
```diff
@@ -405,11 +401,14 @@ def mcmc(num_warmup, num_samples, init_params, num_chains=1, sampler='hmc',
...
+            rngs = sampler_kwargs.pop('rng', vmap(PRNGKey)(np.arange(num_chains)))
+        else:
+            rng = sampler_kwargs.pop('rng', PRNGKey(0))
 
         init_kernel, sample_kernel = hmc(potential_fn, kinetic_fn, algo)
         if progbar:
-            hmc_state = init_kernel(init_params, num_warmup, progbar=progbar, rng=rngs[0],
+            hmc_state = init_kernel(init_params, num_warmup, progbar=progbar, rng=rng,
                                     **sampler_kwargs)
             samples_flat = fori_collect(0, num_samples, sample_kernel, hmc_state,
```
Comment (neerajprad) [inline numpyro/mcmc.py]: Not related to this PR, but could you add `progbar_desc='sample'` in `single_chain_mcmc`?

File: notebooks/source/bayesian_regression.ipynb [hunk 2]
```diff
@@ -1342,7 +1347,8 @@
     "\n",
     "Let us now look at the posterior predictive distribution to see how our predictive distribution looks with respect to the observed divorce rates. To get samples from the posterior predictive distribution, we need to run the model by substituting the latent parameters with samples from the posterior. This sounds complicated, but this can be easily achieved by using effect handlers from the [handlers module](https://numpyro.readthedocs.io/en/latest/handlers.html).\n",
     "\n",
-    "In particular, note the use of the `substitute`, `seed` and `trace` effect handlers in the `predict` function. \n",
+    "In particular, note the use of the `substitute`, `seed` and `trace` effect handlers in the `predict` function.\n",
+    "\n",
```
Comment (neerajprad) [inline notebooks/source/bayesian_regression.ipynb]: Thanks for fixing this!

File: numpyro/mcmc.py (hunk 1 above)
Comment (fehiepsi) [reply numpyro/mcmc.py]: I just check: that `single_chain_mcmc` does not use `progbar` so I think that adding `progbar_desc='sample'` does not take effect.

Comment (neerajprad) [reply numpyro/mcmc.py]: Sorry, my bad. I saw this in some test, but that uses fori_collect directly, so it isn't related.

**Pull request merged**